tests/20-charm-validation.py

```diff
@@ -90,6 +90,19 @@ def test_worker_services(self):
             for service in services:
                 self.assertTrue(check_systemd_service(worker, service))
 
+    def test_scale_master_services(self):
+        '''Test we can scale kubernetes masters.'''
+        for master in self.masters:
+            self.deployment.remove_unit(master.info['unit_name'])
+        self.deployment.sentry.wait()
+        assert len(self.masters) is 0
+
+        self.deployment.add_unit('kubernetes-master', 2)
+        self.deployment.sentry.wait()
+        for master in self.masters:
+            output, rc = master.run('grep server: /home/ubuntu/config')
+            self.assertTrue(rc == 0)
```

Comment:
I see that you add 2 masters here but dont check if the IP matches the recently added units. The test could potentially report a false positive as we aren't checking for the server: key when there are no master units. So what happens if the config just isn't updated? or what if it has the older ip address from the original server? There are two scenarios I can think of that would cause this to pass when the behavior is actually bad.